charts/datadog/templates/cluster-agent-rbac.yaml

```
@@ -303,6 +233,23 @@ subjects:
     name: {{ template "datadog.fullname" . }}-cluster-agent
     namespace: {{ .Release.Namespace }}
 ---
+{{- if and .Values.datadog.kubeStateMetricsCore.enabled (not .Values.datadog.kubeStateMetricsCore.useClusterCheckWorkers) }}
```

That’s really a tiny detail, but as `---` is the markdown marker for the beginning of a document, it makes more sense to include the leading `---` in the conditional `{{- if` block.
(The markdown marker for the end of a document is `...` but is usually omitted.)

```suggestion
{{- if and .Values.datadog.kubeStateMetricsCore.enabled (not .Values.datadog.kubeStateMetricsCore.useClusterCheckWorkers) }}
---
```